AccessControlException is thrown when the security manager is enabled

I was trying to run this code (based on [HttpRealWorldTest](https://github.com/scalaj/scalaj-http/blob/master/src/test/scala/scalaj/http/HttpRealWorldTest.scala)):
```scala
package shuttlecraft

import scalaj.http._

object Main {
  def main(args: Array[String]): Unit = {
    println("Hello, world!")
    val response = Http("https://twitter.com").asString
    println(response)
  }
}
```
It runs without error when launched with a plain `java` command, but fails with `-Djava.security.manager` enabled. 

Here is the stack trace I'm getting:
```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at scalaj.http.BaseHttp$.$lessinit$greater$default$2(Http.scala:795)
	at scalaj.http.Http$.<init>(Http.scala:779)
	at scalaj.http.Http$.<clinit>(Http.scala)
	at shuttlecraft.Main$.main(Main.scala:8)
	at shuttlecraft.Main.main(Main.scala)
Caused by: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "accessDeclaredMembers")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at java.lang.Class.checkMemberAccess(Class.java:2348)
	at java.lang.Class.getDeclaredMethod(Class.java:2127)
	at scalaj.http.HttpConstants$.liftedTree1$1(Http.scala:675)
	at scalaj.http.HttpConstants$.<init>(Http.scala:674)
	at scalaj.http.HttpConstants$.<clinit>(Http.scala)
	... 5 more
```
It seems that the exception is caused by the attempt to [determine](https://github.com/scalaj/scalaj-http/blame/master/src/main/scala/scalaj/http/Http.scala#L675) with reflection whether [`HttpURLConnection.setFixedLengthStreamingMode`](https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html#setFixedLengthStreamingMode-long-) accepts a parameter of type `Long`, which violates the [accessDeclaredMembers](https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimePermission.html) policy.

--- a/src/main/scala/scalaj/http/Http.scala
+++ b/src/main/scala/scalaj/http/Http.scala
@@ -670,22 +670,7 @@ object HttpConstants {
   )
 
   val setFixedLengthStreamingMode: (HttpURLConnection, Long) => Unit = {
-    val connClass = classOf[HttpURLConnection]
-    val (isLong, theMethod) = try {
-      true -> connClass.getDeclaredMethod("setFixedLengthStreamingMode", java.lang.Long.TYPE)
-    } catch {
-      case e: NoSuchMethodException =>
-        false -> connClass.getDeclaredMethod("setFixedLengthStreamingMode", java.lang.Integer.TYPE)
-    }
-    (conn, length) => 
-      if (isLong) {
-        theMethod.invoke(conn, length: java.lang.Long)
-      } else {
-        if (length > Int.MaxValue) {
-          throw new RuntimeException("Failing attempt to upload file greater than 2GB on java version < 1.7")
-        }
-        theMethod.invoke(conn, length.toInt: java.lang.Integer)
-      }
+    case (connection, contentLength) => connection.setFixedLengthStreamingMode(contentLength)
   }
 
   def urlEncode(name: String, charset: String): String = URLEncoder.encode(name, charset)

